lib/rubycritic/source_control_systems/git/revision.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module RubyCritic
+  module SourceControlSystem
+    class Git < Base
+      class Revision
+        def initialize(path, new_path)
+          @path = path
+          @new_path = new_path
+        end
+
+        def self.matches?(status)
+          match =~ status
+        end
+
+        def self.parse(revisions, string)
+          status, path, new_path = string.split("\t").map(&:shellescape)
+          revisions
+            .find { |revision| revision.matches?(status) }
+            .new(path, new_path)
```

Comment:
@joneshf I think the test errors come from here. I think that in some strange case, the `.matches?` never returns true.

Maybe status is nil, or empty?

Reply:
@joneshf bump! hacktoberfest is just around the corner :wink:

Reply:
Sorry for leaving this open for like four months 😅.

Work is letting up a bit so I can probably jump back on it soon. If I don't get to it within the next two weeks, I'll close it since I probably won't be finishing it up.